vscode/src/ghCopilot.ts

```diff
@@ -0,0 +1,226 @@
+          fail $"DoOp returned {result}"
+      }
+  }
+  \`\`\`
+
+  Note: Prefer using a conditional \`fail\` statement to \`Fact\` calls, as \`fail\` gives a better error location.
+
+  ## Syntax
+
+  When writing Q#, the \`for (i in 1..100)\` syntax is invalid, use \`for i in 1..100\` or
+  \`for element in array[2...]\`.
+
+  While many Q# operators are C-like, it uses \`or\` instead of \`||\` and \`and\` instead of \`&&\`.
+
+  ## Multi-file projects
+
+  Modern Q# does not use \`namespace\` blocks to enclose code.
+  When Q# code is in different files in a project, each function or operation is in a namespace
+  which is the name of the containing file. For example, if \`Main.qs\` has an operation \`Foo\`,
+  then \`Tests.qs\` could reference the operation as \`Main.Foo\`, or bring \`Foo\` into scope by
+  adding \`import Main.Foo;\` in the file.
+
+  ## Libraries
+
+  A Q# project can reference a library from GitHub but updating the \`dependencies\` entry of
+  the \`qsharp.json\` file. For example, to reference the \`chemistry\` library, the \`qsharp.json\`
+  file might appear as:
+
+  \`\`\`json
+  {
+      "dependencies": {
+          "Chemistry": {
+              "github": {
+                  "ref": "v1.15.0",
+                  "owner": "microsoft",
+                  "repo": "qsharp",
+                  "path": "library/chemistry"
+              }
+          }
+      }
+  }
+  \`\`\`
+
+  ## Jupyter Notebooks
+
+  Q# has first-class support for Jupyter notebooks. Typically the first cell will contain \`import qsharp\`.
+
+  Jupyter cells can contain Q# code directly by using the \`%%qsharp\` magic command at the beginning of the cell. For example:
+
+  \`\`\`python
+  %%qsharp
+
+  operation GHZSample(n: Int) : Result[] {
+      use qs = Qubit[n];
+
+      H(qs[0]);
+      ApplyToEach(CNOT(qs[0], _), qs[1...]);
+
+      let results = MeasureEachZ(qs);
+      ResetAll(qs);
+      return results;
+  }
+  \`\`\`
+
+  The \`qsharp_widgets\` package provides viewers for circuits and histograms, e.g.
+
+  \`\`\`python
+  from qsharp_widgets import Circuit, Histogram
+  Circuit(qsharp.circuit("GHZSample(3)"))
+  \`\`\`
+
+  Note that the latest Q# and QDK releases don't require or use the old IQ# kernel. It just needs to the \`qsharp\` PyPI package,
+  and maybe \`qsharp_widgets\` for visuals.`;
+
+/**
+ * Checks if the copilot-instructions.md file exists and contains Q# instructions.
+ *
+ * @param workspaceFolder The workspace folder URI.
+ * @returns A promise that resolves to true if the file exists and contains Q# instructions.
+ */
+async function hasQSharpCopilotInstructions(
+  workspaceFolder: vscode.Uri,
+): Promise<boolean> {
+  try {
+    const githubDir = vscode.Uri.joinPath(workspaceFolder, ".github");
+    const instructionsFile = vscode.Uri.joinPath(
+      githubDir,
+      "copilot-instructions.md",
+    );
+
+    // Check if file exists
+    await vscode.workspace.fs.stat(instructionsFile);
```

Comment:
Should it do something with the result of this? Seems like the below should be wrapped in an `if` block conditional on this check.

Reply:
(Or does it just throw, hence the catch below?)